pkg/runtime/core/cloneable.go

```diff
@@ -0,0 +1,16 @@
+package core
+
+var cloneableTypes = map[Type]bool{
+	ArrayType:  true,
+	ObjectType: true,
+}
+
+type Cloneable interface {
+	Value
+	Clone() Cloneable
+}
+
+func IsCloneable(value Value) bool {
```

Comment:
Could you move this helper function to ``values`` package? It's very implementation sensitive code.

Reply:
What about make```IsCloneable(value values.Value) values.Boolean```?

Reply:
@3timeslazy I do not have a strong opinion on this since it's for internal use only. 
It's up to you :)